Source/SwiftLintBuiltInRules/Rules/Lint/PrivateSwiftUIStatePropertyRule.swift

```diff
@@ -202,58 +32,170 @@ private extension PrivateSwiftUIStatePropertyRule {
         }
 
         /// LIFO stack that stores type inheritance clauses for each visited node
-        /// The last value is the inheritance clause for the most recently visited node
-        /// A nil value indicates that the node does not provide any inheritance clause
-        private var visitedTypeInheritances = Stack<InheritanceClauseSyntax?>()
+        /// `true` indicates the the most recently visited decl syntax 
+        /// should be evaluated for SwiftUI State access levels
+        private var visitedScopes = Stack<Bool>()
 
         override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
-            visitedTypeInheritances.push(node.inheritanceClause)
+            visitedScopes.push(node.inheritanceClause.conformsToApplicableSwiftUIProtocol)
             return .visitChildren
         }
 
         override func visitPost(_ node: ClassDeclSyntax) {
-            visitedTypeInheritances.pop()
+            visitedScopes.pop()
         }
 
         override func visit(_ node: StructDeclSyntax) -> SyntaxVisitorContinueKind {
-            visitedTypeInheritances.push(node.inheritanceClause)
+            visitedScopes.push(node.inheritanceClause.conformsToApplicableSwiftUIProtocol)
             return .visitChildren
         }
 
         override func visitPost(_ node: StructDeclSyntax) {
-            visitedTypeInheritances.pop()
+            visitedScopes.pop()
         }
 
         override func visit(_ node: ActorDeclSyntax) -> SyntaxVisitorContinueKind {
-            visitedTypeInheritances.push(node.inheritanceClause)
+            visitedScopes.push(node.inheritanceClause.conformsToApplicableSwiftUIProtocol)
             return .visitChildren
         }
 
         override func visitPost(_ node: ActorDeclSyntax) {
-            visitedTypeInheritances.pop()
+            visitedScopes.pop()
+        }
+
+        override func visit(_ node: VariableDeclSyntax) -> SyntaxVisitorContinueKind {
+            guard let parent = node.parent, parent.is(MemberBlockItemSyntax.self) else {
+                return .skipChildren
+            }
+
+            return .visitChildren
         }
 
-        override func visitPost(_ node: MemberBlockItemSyntax) {
+        override func visitPost(_ node: VariableDeclSyntax) {
             guard
-                let decl = node.decl.as(VariableDeclSyntax.self),
-                let inheritanceClause = visitedTypeInheritances.peek() as? InheritanceClauseSyntax,
-                inheritanceClause.conformsToApplicableSwiftUIProtocol,
-                decl.attributes.hasStateAttribute,
-                !decl.modifiers.containsPrivateOrFileprivate()
+                let shouldCheckAccessLevel = visitedScopes.peek(),
+                shouldCheckAccessLevel,
+                node.containsSwiftUIStateAccessLevelViolation
```

Comment:
```suggestion
                visitedScopes.peek(),
                node.containsSwiftUIStateAccessLevelViolation
```

Reply:
`peek()` returns an optional value, so I'd have to nil coalesce to a default value to get this to compile. I could nil coalesce to `false` to get it down to a single line if that's preferable, though.